src/seaport.ts

```diff
@@ -1746,8 +1809,17 @@ export class OpenSeaPort {
         ? makeBigNumber(sellOrder.takerRelayerFee)
         : makeBigNumber(sellOrder.makerRelayerFee)
     } else {
+      // If all assets are for the same contract and it's a non-private sale, use its fees
+      let asset: OpenSeaAsset | null = null
+      if (assets && _.uniqBy(assets, a => a.tokenAddress).length == 1) {
+        const { tokenAddress, tokenId } = assets[0]
+        asset = await this.api.getAsset(tokenAddress, tokenId)
```

Comment:
let's change the type of getAsset to OpenSeaAsset (nonnull), and remove the json null check there. that way you won't have to check for null all over the place here. 404s are thrown as errors by Transport.ts